lib/Viroverse/Controller/Sample.pm

```diff
@@ -160,6 +162,164 @@ sub ice_cultures : Chained('page_base') PathPart('ice-cultures') Args(0) {
+        bold        => 1,
+        color       => 'white',
+    );
+
+    my $row = 0;
+
+    my $add_row = sub {
+        my ($key, $name, $value, $value_format) = @_;
+
+        $sheet->write($row, 0, $key,    $label);
+        $sheet->write($row, 1, $name,   $label);
+        $sheet->write($row, 2, $value,  $value_format);
+        ++$row;
+    };
+
+    $sheet->write($row, 0, 'start_metadata',    $header);
+    $sheet->write($row, 1, 'Name',              $header);
+    $sheet->write($row, 2, 'Value',             $header);
+    ++$row;
+
+    $add_row->('vv_sample_id',      'Viroverse ID',     $sample_id, $locked);
+    $add_row->('pcr_date_round1',   'PCR Round 1 Date', undef,      $unlocked);
+    $add_row->('pcr_date_round2',   'PCR Round 2 Date', undef,      $unlocked);
+    $add_row->('pcr_date_round3',   'PCR Round 3 Date', undef,      $unlocked);
+
+    $add_row->();
+
+    $add_row->('sample_type',      'Sample Type',  undef,      $unlocked);
+    $add_row->('pcr_reaction',     'PCR Reaction', undef,      $unlocked);
+    $add_row->('primer',           'Primer',       undef,      $unlocked);
+    $add_row->('volume',           'µL/rxn',       undef,      $unlocked);
+    $add_row->('dna_mass',         'µg DNA/rxn',   undef,      $unlocked);
+    $add_row->('notes',            'Notes',        undef,      $unlocked);
+
+
+    $add_row->('end_metadata');
+
+    $sheet->write($row, 0, "start_pcr",     $header);
+    $sheet->write($row, 1, "Well Number",   $header);
+    $sheet->write($row, 2, "Position",           $header);
+    $sheet->write($row, 3, "Sample ID",     $header);
+    $sheet->write($row, 4, "+/-",           $header);
+    ++$row;
+
+    my $add_pcr_well = sub {
+        my ($well_num, $sample_name) = @_;
+        my $well_name = Viroverse::Model::gel_lane->intTo96Well($well_num);
+
+        $sheet->write($row, 0, "pcr_well",      $label);
+        $sheet->write($row, 1, $well_num,       $label);
+        $sheet->write($row, 2, $well_name,      $label);
+        $sheet->write($row, 3, $sample_name,    $unlocked);
+        $sheet->write($row, 4, undef,           $unlocked);
+        ++$row;
+    };
+
+    my $first_pcr_row = $row;
+
+    for my $well (1..96) {
+        $add_pcr_well->($well, undef);
+    }
+
+    my $last_pcr_row = $row - 1;
+
+    $add_row->('end_pcr');
+
+    $sheet->conditional_formatting($first_pcr_row, 4, $last_pcr_row, 4, {
+        type => 'text',
+        criteria => 'begins with',
+        value => '+',
+        format => $pcr_positive,
+        stop_if_true => 1,
+    });
+    $sheet->conditional_formatting($first_pcr_row, 4, $last_pcr_row, 4, {
+        type => 'text',
+        criteria => 'begins with',
+        value => '-',
+        format => $pcr_negative,
+        stop_if_true => 1,
+    });
+    $sheet->conditional_formatting($first_pcr_row, 4, $last_pcr_row, 4, {
+        type => 'no_blanks',
+        format => $pcr_error,
+        stop_if_true => 1,
+    });
+
+    $xls->close();
```

Comment:
Can return falsy indicating an error without dying, [see](https://metacpan.org/pod/Excel::Writer::XLSX#close()). Please add `or die $some_error` instead of potentially sending a garbage response.